packages/vue/__tests__/Transition.spec.ts

```diff
@@ -1325,6 +1325,69 @@ describe('e2e: Transition', () => {
       },
       E2E_TIMEOUT
     )
+
+    test(
+      'Suspense fallback should work with transition',
+      async () => {
+        const onLeaveSpy = jest.fn()
+        const onEnterSpy = jest.fn()
+
+        await page().exposeFunction('onLeaveSpy', onLeaveSpy)
+        await page().exposeFunction('onEnterSpy', onEnterSpy)
+
+        await page().evaluate(() => {
+          const { createApp, shallowRef, h } = (window as any).Vue
+          const One = {
+            template: `<div>{{ msg }}</div>`,
+            setup() {
+              return new Promise((resolve, reject) => {
+                setTimeout(
+                  () =>
+                    resolve({
+                      msg: 'success'
+                    }),
+                  1000
+                )
+              })
+            }
+          }
+          createApp({
+            template: `
+              <div id="container">
+                <transition mode="out-in">
+                  <Suspense :timeout="0">
+                    <template #default>
+                      <component :is="view" />
+                    </template>
+                    <template #fallback>
+                      <div>Loading...</div>
+                    </template>
+                  </Suspense>
+                </transition>
+              </div>
+              <button id="toggleBtn" @click="click">button</button>
+            `,
+            setup: () => {
+              const view = shallowRef(null)
+              const click = () => {
+                view.value = view.value ? null : h(One)
+              }
+              return { view, click }
+            }
+          }).mount('#app')
+        })
+
+        await nextTick()
+        expect(await html('#container')).toBe('<!---->')
+
+        await click('#toggleBtn')
+        await timeout(500)
+        expect(await html('#container')).toBe('<div class="">Loading...</div>')
+        await transitionFinish(1500)
```

Comment:
Avoid using hard coded test durations like this.

See https://github.com/vuejs/vue-next/pull/3968/commits/740bcd17141c72f14af0ef8b15601e1a7f0312de